internal/state/statuses.go

```diff
@@ -127,3 +137,16 @@ func buildStatuses(graph *graph) Statuses {
 
 	return statuses
 }
+
+func buildBaseRouteConditions(gcValidAndExist bool) []conditions.RouteCondition {
```

Comment:
Is there a way to avoid passing a boolean as an argument here? I'd like to avoid boolean arguments whenever possible because it implies that the function does two things.

Reply:
I expect this code to be changed, because of the FIXME below:
```
	// FIXME(pleshakov): Figure out appropriate conditions for the cases when:
	// (1) GatewayClass is invalid.
	// (2) GatewayClass does not exist.
```

So  we'll need to handle those cases separately, for which a single boolean condition will no longer work.

Considering that, do you think it still makes sense to refactor?

Reply:
Ah, ok. Probably not.